Mathlib/Algebra/CharP/Basic.lean

```diff
@@ -540,6 +540,12 @@ theorem char_is_prime_or_zero (p : ℕ) [hc : CharP R p] : Nat.Prime p ∨ p = 0
   | m + 2, hc => Or.inl (@char_is_prime_of_two_le R _ _ (m + 2) hc (Nat.le_add_left 2 m))
 #align char_p.char_is_prime_or_zero CharP.char_is_prime_or_zero
 
+theorem exists' (R : Type*) [NonAssocRing R] [NoZeroDivisors R] [Nontrivial R] :
```

Comment:
`ExpChar.exists` all uses domain, so let's use it here too:
```suggestion
theorem exists' (R : Type*) [NonAssocRing R] [IsDomain R] :
```

Reply:
No, it doesn't work as `IsDomain R` requires `R` to be an associated ring.